rust/worker/src/cache/cache.rs

```diff
@@ -0,0 +1,190 @@
+                        Some(value)
+                    }
+                    None => None,
+                }
+            }
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct UnboundedCache<K, V>
+where
+    K: Send + Sync + Hash + Eq + 'static,
+    V: Send + Sync + Clone + 'static,
+{
+    cache: Arc<RwLock<HashMap<K, V>>>,
+}
+
+impl<K, V> UnboundedCache<K, V>
+where
+    K: Send + Sync + Hash + Eq + 'static,
+    V: Send + Sync + Clone + 'static,
+{
+    pub fn new(config: &CacheConfig) -> Self {
+        match config {
+            CacheConfig::Unbounded(_) => UnboundedCache {
+                cache: Arc::new(RwLock::new(HashMap::new())),
+            },
+            _ => panic!("Invalid cache configuration"),
+        }
+    }
+
+    pub fn insert(&self, key: K, value: V) {
+        self.cache.write().insert(key, value);
+    }
+
+    pub fn get(&self, key: &K) -> Option<V> {
+        let read_guard = self.cache.read();
+        let value = read_guard.get(key);
+        match value {
+            Some(v) => Some(v.clone()),
+            None => None,
+        }
+    }
+}
+
+pub struct FoyerCacheWrapper<K, V>
+where
+    K: Send + Sync + Hash + Eq + 'static,
+    V: Send + Sync + Clone + 'static,
+{
+    cache: FoyerCache<K, V>,
+}
+
+impl<K, V> FoyerCacheWrapper<K, V>
+where
+    K: Send + Sync + Hash + Eq + 'static,
+    V: Send + Sync + Clone + 'static,
+{
+    pub fn new(config: &CacheConfig) -> Self {
+        match config {
+            CacheConfig::Lru(lru) => {
+                // TODO: add more eviction config
+                let eviction_config = LruConfig::default();
+                let cache_builder =
+                    CacheBuilder::new(lru.capacity).with_eviction_config(eviction_config);
+                FoyerCacheWrapper {
+                    cache: cache_builder.build(),
+                }
+            }
+            CacheConfig::Lfu(lfu) => {
+                // TODO: add more eviction config
+                let eviction_config = LfuConfig::default();
+                let cache_builder =
+                    CacheBuilder::new(lfu.capacity).with_eviction_config(eviction_config);
+                FoyerCacheWrapper {
+                    cache: cache_builder.build(),
+                }
+            }
+            _ => panic!("Invalid cache configuration"),
+        }
+    }
+
+    pub fn insert(&self, key: K, value: V) {
+        self.cache.insert(key, value);
+    }
+
+    pub fn get(&self, key: &K) -> Option<V> {
+        let entry = self.cache.get(key);
+        match entry {
+            Some(v) => {
+                let value = v.value().to_owned();
+                Some(value)
+            }
+            None => None,
+        }
+    }
+}
+
+#[async_trait]
+impl<K, V> Configurable<CacheConfig> for UnboundedCache<K, V>
+where
+    K: Send + Sync + Hash + Eq + 'static,
+    V: Send + Sync + Clone + 'static,
+{
+    async fn try_from_config(config: &CacheConfig) -> Result<Self, Box<dyn ChromaError>> {
```

Comment:
There should not be an implementation for UnboundedCache and for FoyerCacheWrapper, instead I think we should have impl Configurable for Cache<K, V> and that will return that higher-level enum.